great_tables/_locations.py

```diff
@@ -355,15 +366,38 @@ def resolve(loc: Loc, *args: Any, **kwargs: Any) -> Loc | list[CellPos]:
 
 
 @resolve.register
-def _(loc: LocColumnSpanners, spanners: Spanners) -> LocColumnSpanners:
+def _(loc: LocSpannerLabels, spanners: Spanners) -> LocSpannerLabels:
     # unique labels (with order preserved)
     spanner_ids = [span.spanner_id for span in spanners]
 
     resolved_spanners_idx = resolve_vector_i(loc.ids, spanner_ids, item_label="spanner")
     resolved_spanners = [spanner_ids[idx] for idx in resolved_spanners_idx]
 
     # Create a list object
-    return LocColumnSpanners(ids=resolved_spanners)
+    return LocSpannerLabels(ids=resolved_spanners)
+
+
+@resolve.register
+def _(loc: LocColumnLabels, data: GTData) -> list[CellPos]:
+    cols = resolve_cols_i(data=data, expr=loc.columns)
+    cell_pos = [CellPos(col[1], 0, colname=col[0]) for col in cols]
+    return cell_pos
+
+
+@resolve.register
+def _(loc: LocRowGroups, data: GTData) -> set[int]:
+    # TODO: what are the rules for matching row groups?
+    # TODO: resolve_rows_i will match a list expr to row names (not group names)
+    group_pos = set(pos for _, pos in resolve_rows_i(data, loc.rows))
```

Comment:
Note that this currently can't match row groups by their name (which is also their id). It can only match by index number or match every group.

Reply:
Totally okay since we don't currently make it possible for the user to set nice IDs anyhow.